ed_sensor_integration/src/laser/laser_plugin.cpp

```diff
@@ -21,240 +22,46 @@
 
 #include <tue/profiling/timer.h>
 
-namespace
-{
-
-typedef std::vector<unsigned int> ScanSegment;
 
-struct EntityUpdate
+/**
+ * copy a ros laserscan message to a standard vector
+ *
+ * @param[in] scan laserscan message
+ * @param[out] sensor_ranges vector with ranges
+ */
+void lasermsgToSensorRanges(const sensor_msgs::LaserScan::ConstPtr& scan, std::vector<float>& sensor_ranges)
 {
-    ed::ConvexHull chull;
-    geo::Pose3D pose;
-    std::string flag; // Temp for RoboCup 2015; todo: remove after
-};
+    if (sensor_ranges.size() != scan->ranges.size())
+        sensor_ranges.resize(scan->ranges.size());
 
-// ----------------------------------------------------------------------------------------------------
-
-double getFittingError(const ed::Entity& e, const geo::LaserRangeFinder& lrf, const geo::Pose3D& rel_pose,
-                       const std::vector<float>& sensor_ranges, const std::vector<double>& model_ranges,
-                       int& num_model_points)
-{
-    std::vector<double> test_model_ranges = model_ranges;
-
-    // Render the entity with the given relative pose
-    geo::LaserRangeFinder::RenderOptions opt;
-    opt.setMesh(e.shape()->getMesh(), rel_pose);
-
-    geo::LaserRangeFinder::RenderResult res(test_model_ranges);
-    lrf.render(opt, res);
-
-    int n = 0;
-    num_model_points = 0;
-    double total_error = 0;
-    for(unsigned int i = 0; i < test_model_ranges.size(); ++i)
+    for(unsigned int i = 0; i < scan->ranges.size(); ++i)
     {
-        double ds = sensor_ranges[i];
-        double dm = test_model_ranges[i];
-
-        if (ds <= 0)
-            continue;
-
-        ++n;
-
-        if (dm <= 0)
-        {
-            total_error += 0.1;
-            continue;
-        }
-
-        double diff = std::abs(ds - dm);
-        if (diff < 0.1)
-            total_error += diff;
+        float r = scan->ranges[i];
+        if (r > scan->range_max)
+            sensor_ranges[i] = r;
+        else if (r == r && r > scan->range_min)
```

Comment:
`r == r` is always true....

Reply:
https://stackoverflow.com/questions/570669/checking-if-a-double-or-float-is-nan-in-c
It is a way of checking if a double is NaN. But some compilers may optimize it away...